crates/uv/tests/common/mod.rs

```diff
@@ -56,29 +57,75 @@ impl TestContext {
         let cache_dir = assert_fs::TempDir::new().expect("Failed to create cache dir");
         let venv = create_venv(&temp_dir, &cache_dir, python_version);
 
+        // The workspace root directory is not available without walking up the tree
+        // https://github.com/rust-lang/cargo/issues/3946
+        let workspace_root = Path::new(&std::env::var("CARGO_MANIFEST_DIR").unwrap())
+            .join("..")
+            .join("..");
+
+        let site_packages = site_packages_path(&venv, format!("python{python_version}"));
+
         let mut filters = Vec::new();
         filters.extend(
             Self::path_patterns(&cache_dir)
                 .into_iter()
                 .map(|pattern| (pattern, "[CACHE_DIR]/".to_string())),
         );
         filters.extend(
-            Self::path_patterns(&temp_dir)
+            Self::path_patterns(&site_packages)
                 .into_iter()
-                .map(|pattern| (pattern, "[TEMP_DIR]/".to_string())),
+                .map(|pattern| (pattern, "[SITE_PACKAGES]/".to_string())),
         );
         filters.extend(
             Self::path_patterns(&venv)
                 .into_iter()
                 .map(|pattern| (pattern, "[VENV]/".to_string())),
         );
+        filters.extend(
+            Self::path_patterns(&temp_dir)
+                .into_iter()
+                .map(|pattern| (pattern, "[TEMP_DIR]/".to_string())),
+        );
+        filters.extend(
+            Self::path_patterns(&workspace_root)
+                .into_iter()
+                .map(|pattern| (pattern, "[WORKSPACE]/".to_string())),
+        );
+
+        // Account for [`Simplified::user_display`] which is relative to the command working directory
+        filters.push((
+            Self::path_pattern(
+                site_packages
+                    .strip_prefix(&temp_dir)
+                    .expect("The test site-packages directory is always in the tempdir"),
+            ),
+            "[SITE_PACKAGES]/".to_string(),
+        ));
+        filters.push((
+            Self::path_pattern(
+                venv.strip_prefix(&temp_dir)
+                    .expect("The test virtual environment directory is always in the tempdir"),
+            ),
+            "[VENV]/".to_string(),
+        ));
```

Comment:
These new filters account for the changes in #2559 — the test commands should always be run with `temp_dir` as the current directory.